extensions/src/main/java/org/pentaho/platform/workitem/WorkItemLifecycleEventFileWriter.java

```diff
@@ -0,0 +1,199 @@
+/*!
+ * This program is free software; you can redistribute it and/or modify it under the
+ * terms of the GNU Lesser General Public License, version 2.1 as published by the Free Software
+ * Foundation.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along with this
+ * program; if not, you can obtain a copy at http://www.gnu.org/licenses/old-licenses/lgpl-2.1.html
+ * or from the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details.
+ *
+ * Copyright (c) 2017 Pentaho Corporation. All rights reserved.
+ */
+
+package org.pentaho.platform.workitem;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import groovy.text.SimpleTemplateEngine;
+import groovy.text.TemplateEngine;
+import org.apache.commons.collections4.CollectionUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.log4j.DailyRollingFileAppender;
+import org.apache.log4j.Level;
+import org.apache.log4j.Logger;
+import org.apache.log4j.PatternLayout;
+import org.pentaho.platform.workitem.messages.Messages;
+import org.springframework.context.event.EventListener;
+import org.springframework.scheduling.annotation.Async;
+import org.springframework.stereotype.Component;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This component listens for {@link WorkItemLifecycleEvent}s and writes the information stored within the event into
+ * a log file.
+ */
+@Component
+public class WorkItemLifecycleEventFileWriter {
+
+  private static final Log log = LogFactory.getLog( WorkItemLifecycleEventFileWriter.class );
+
+  private static String WORK_ITEM_LOG_FILE = "work-item-status";
+
+  private static final String DEFAULT_FIELD_DELIMITER = "|";
+
+  private static List<String> DEFAULT_EVENT_MESSAGE_FIELDS = new ArrayList<String>();
+  static {
+    DEFAULT_EVENT_MESSAGE_FIELDS.add( "targetTimestamp" );
+    DEFAULT_EVENT_MESSAGE_FIELDS.add( "workItemUid" );
+    DEFAULT_EVENT_MESSAGE_FIELDS.add( "workItemLifecyclePhase" );
+    DEFAULT_EVENT_MESSAGE_FIELDS.add( "lifecycleDetails" );
+    DEFAULT_EVENT_MESSAGE_FIELDS.add( "sourceTimestamp" );
+    DEFAULT_EVENT_MESSAGE_FIELDS.add( "sourceHostName" );
+    DEFAULT_EVENT_MESSAGE_FIELDS.add( "sourceHostIp" );
+    DEFAULT_EVENT_MESSAGE_FIELDS.add( "workItemDetails" );
+  }
+
+  private static List<String> DEFAULT_ENV_VAR_MESSAGE_FIELDS = new ArrayList<String>();
+  static {
+    DEFAULT_ENV_VAR_MESSAGE_FIELDS.add( "HOSTNAME" );
```

Comment:
At some point we need to make this declarative so these vars can be easily changed depending on the orchestration environ.  We will likely need Chronos specific ones soon.